Websites/webkit.org/demos/webgpu/scripts/hello-triangle.js

```diff
@@ -28,24 +28,33 @@ async function helloTriangle() {
     const pipelineLayoutDesc = { bindGroupLayouts: [uniformBindGroupLayout] };
     const layout = device.createPipelineLayout(pipelineLayoutDesc);
 /*
-    FIXME: Use WGSL once compiler is brought up
+    FIXME: Add support for binary expressions to match the metal code
+*/
     const wgslSource = `
-                     @vertex fn vsmain(@builtin(vertex_index) VertexIndex: u32) -> @builtin(position) vec4<f32>
+                     struct Vertex {
+                         @builtin(position) Position: vec4<f32>,
+                         @location(0) color: vec4<f32>,
+                     }
+
+                     @vertex fn vsmain(@builtin(vertex_index) VertexIndex: u32) -> Vertex
                      {
                          var pos: array<vec2<f32>, 3> = array<vec2<f32>, 3>(
                              vec2<f32>( 0.0,  0.5),
                              vec2<f32>(-0.5, -0.5),
                              vec2<f32>( 0.5, -0.5)
                          );
-                         return vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+                         var vout : Vertex;
```

Comment:
`vout`? Usually we try to use full words

Reply:
sure, I  can rename, but I just copied this from a couple lines below.